src/client/js/partials/breaches.js

```diff
@@ -141,12 +143,55 @@ function renderZeroState () {
   breachesTable.append(content)
 }
 
+function renderPieChart () {
+  const rowsForSelectedEmail = Array.from(breachesTable.querySelectorAll(`[data-email='${state.selectedEmail}']`))
+  const classesForSelectedEmail = rowsForSelectedEmail.flatMap(row => row.dataset.classes.split(','))
+  const classesMap = classesForSelectedEmail.reduce((acc, cur) => {
+    acc.set(cur, (acc.get(cur) ?? 0) + 1) // set count for each class key
+    return acc
+  }, new Map())
+  const classesTop3 = [...classesMap.keys()].sort((a, b) => classesMap.get(b) - classesMap.get(a)).slice(0, 3)
+  const classesTotal = classesForSelectedEmail.length
+  const chartData = []
+
+  switch (true) {
+    case classesMap.size === 0:
+      chartData.push({
+        key: pieChart.dataset.txtNone,
+        name: pieChart.dataset.txtNone,
+        count: 1,
+        color: chartColors[4]
+      })
+      break
+    case classesMap.size >= 4:
+      chartData[3] = {
+        key: pieChart.dataset.txtOther,
+        name: pieChart.dataset.txtOther,
+        count: classesTotal - classesMap.get(classesTop3[0]) - classesMap.get(classesTop3[1]) - classesMap.get(classesTop3[2]),
+        color: chartColors[3]
+      }
+      // falls through
+    default:
+      classesTop3.forEach((name, i) => {
+        chartData[i] = {
+          key: name,
+          name,
+          count: classesMap.get(name),
+          color: chartColors[i]
```

Comment:
I'm guessing the order will generally stay the same, so in practice this might not be relevant, but I was wondering if it might not be helpful if the same type of data would always get the same colour? Otherwise, every time the user switches between different email addresses, they'll have to consult the legend again.